src/vmm/src/config/builder.rs

```diff
@@ -16,17 +16,43 @@ pub struct Builder {
 
 impl Default for Builder {
     fn default() -> Self {
-        Self::new()
+        Builder {
+            inner: Ok(VMMConfig::default()),
+        }
     }
 }
 
 impl Builder {
+    /// Creates a Builder Object
+    ///
     pub fn new() -> Self {
-        Builder {
-            inner: Ok(VMMConfig::default()),
-        }
+        Builder::default()
     }
 
+    /// Builds VMMConfig
+    ///
+    /// This function should be last of the chain to build VMMConfiguration. Since, it is not known
+    /// which of the `config` functions user has called, missing configuration errors are handled
+    /// here -
+    ///  * Empty Kernel Configuration ('path' missing.)
```

Comment:
This comment reveals implementation details. How about we just keep the format typically used by Rust crates and have a short description, then an error section?

For example:
```rust
/// Builds a `VMMConfig`. This function needs to be called after all needed configurations are set.
///
/// # Errors
///
/// Returns an error when the required fields were not set using the appropriate `*_config` functions.
```

This way we do not explicitly mention the required fields, which means that if we add other in the future, this comment will not be outdated.